File: distribution/certification-src/bun/certify.js
const op = Bun.argv[2];
if (op === "stdin") await Bun.write(Bun.stdout, await Bun.stdin.bytes());
else if (op === "exit") process.exit(7);
else if (op === "sleep") await Bun.sleep(5000);
else if (op === "certify") {
  const result = {
    input: await Bun.file(`${process.env.COMPUTE_WORK_DIR}/hello.txt`).text(),
    success: true,
    argument: Bun.argv[3],
    environment: process.env.CERTIFICATION_ENV ?? "missing",
    host_environment: process.env.COMPUTE_HOST_SECRET ?? "missing",
  };
  await Bun.write(`${process.env.COMPUTE_OUTPUT_DIR}/result.json`, JSON.stringify(result));
  console.log(JSON.stringify({ runtime: "bun", runtime_version: Bun.version }));
  console.error("certification-stderr");
}

File: distribution/certification-src/compiled/certify.rs
use std::env;
use std::fs;
use std::io::{self, Read};
use std::process;
use std::thread;
use std::time::Duration;

fn main() {
    let args: Vec<String> = env::args().collect();
    match args[1].as_str() {
        "stdin" => {
            let mut data = Vec::new();
            io::stdin().read_to_end(&mut data).unwrap();
            std::io::Write::write_all(&mut io::stdout(), &data).unwrap();
        }
        "exit" => process::exit(7),
        "sleep" => thread::sleep(Duration::from_secs(5)),
        "certify" => {
            let value = |name: &str| env::var(name).unwrap_or_else(|_| "missing".into());
            let input = fs::read_to_string(format!("{}/hello.txt", value("COMPUTE_WORK_DIR"))).unwrap();
            let result = format!(
                "{{\"input\":\"{}\",\"success\":true,\"argument\":\"{}\",\"environment\":\"{}\",\"host_environment\":\"{}\"}}",
                input, args[2], value("CERTIFICATION_ENV"), value("COMPUTE_HOST_SECRET")
            );
            fs::write(format!("{}/result.json", value("COMPUTE_OUTPUT_DIR")), result).unwrap();
            println!(
                "{{\"runtime\":\"{}\",\"runtime_version\":\"{}\"}}",
                env!("CERTIFICATION_RUNTIME"),
                env!("CERTIFICATION_VERSION")
            );
            eprintln!("certification-stderr");
        }
        _ => process::exit(2),
    }
}
